cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/worker/TaskWorkerHttpHandlerInternal.java

```diff
@@ -73,14 +81,16 @@ public void run(FullHttpRequest request, HttpResponder responder) {
         GSON.fromJson(request.content().toString(StandardCharsets.UTF_8), RunnableTaskRequest.class);
       className = runnableTaskRequest.getClassName();
       byte[] response = runnableTaskLauncher.launchRunnableTask(runnableTaskRequest);
-      responder.sendByteArray(HttpResponseStatus.OK, response, EmptyHttpHeaders.INSTANCE);
+
+      responder.sendContent(HttpResponseStatus.OK,
+                            new RunnableTaskBodyProducer(response, stopper, className),
+                            new DefaultHttpHeaders().add(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_OCTET_STREAM));
     } catch (ClassNotFoundException | ClassCastException ex) {
       responder.sendString(HttpResponseStatus.BAD_REQUEST, exceptionToJson(ex), EmptyHttpHeaders.INSTANCE);
     } catch (Exception ex) {
       LOG.error(String.format("Failed to run task %s",
-                              request.content().toString(StandardCharsets.UTF_8), ex));
+                              request.content().toString(StandardCharsets.UTF_8)), ex);
```

Comment:
nit: are you sure it's a good idea to log whole request that has serialized task within json? I can be a lot of data.

Reply:
Good point. Thx.